tests/framework/extensions/rancher/setup.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/rancher/rancher/tests/framework/extensions/token"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	nodepools "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+	aws "github.com/rancher/rancher/tests/framework/extensions/rke1/nodetemplates/aws"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	clusterName = "local"
+)
+
+func CreateAdminToken(password string, rancherConfig *rancher.Config) (string, error) {
+	adminUser := &management.User{
+		Username: "admin",
+		Password: password,
+	}
+
+	hostURL := rancherConfig.Host
+	var userToken *management.Token
+	err := kwait.Poll(500*time.Millisecond, 5*time.Minute, func() (done bool, err error) {
+		userToken, err = token.GenerateUserToken(adminUser, hostURL)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	return userToken.Token, nil
+}
+
+func PostRancherInstall(adminClient *rancher.Client, adminPassword string) error {
+	clusterID, err := clusters.GetClusterIDByName(adminClient, clusterName)
+	if err != nil {
+		return err
+	}
+
+	steveClient, err := adminClient.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		return err
+	}
+
+	timeStamp := time.Now().Format(time.RFC3339)
+	settingEULA := v3.Setting{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "eula-agreed",
+		},
+		Default: timeStamp,
+		Value:   timeStamp,
+	}
+
+	urlSetting := &v3.Setting{}
+
+	_, err = steveClient.SteveType("management.cattle.io.setting").Create(settingEULA)
+	if err != nil {
+		return err
+	}
+
+	urlSettingResp, err := steveClient.SteveType("management.cattle.io.setting").ByID("server-url")
+	if err != nil {
+		return err
+	}
+
+	err = v1.ConvertToK8sType(urlSettingResp.JSONResp, urlSetting)
+	if err != nil {
+		return err
+	}
+
+	urlSetting.Value = fmt.Sprintf("https://%s", adminClient.RancherConfig.Host)
+
+	_, err = steveClient.SteveType("management.cattle.io.setting").Update(urlSettingResp, urlSetting)
+	if err != nil {
+		return err
+	}
+
+	userList, err := adminClient.Management.User.List(&types.ListOpts{
+		Filters: map[string]interface{}{
+			"username": "admin",
+		},
+	})
+	if err != nil {
+		return err
+	} else if len(userList.Data) == 0 {
+		return fmt.Errorf("admin user not found")
+	}
+
+	adminUser := &userList.Data[0]
+	setPasswordInput := management.SetPasswordInput{
+		NewPassword: adminPassword,
+	}
+	_, err = adminClient.Management.User.ActionSetpassword(adminUser, &setPasswordInput)
+
+	return err
+}
+
+func CreateTestCluster(client, adminClient *rancher.Client, numClusters int, clusterNameBase, cni, kubeVersion string, nodesAndRoles []nodepools.NodeRoles, privateRegistries []management.PrivateRegistry) ([]string, error) {
```

Comment:
So this is specifically an RKE1 cluster, and it relies and `aws.CreateAWSNodeTemplate` which we don't want at all. (I'll change this in code coverage later) We don't ever want to specify the node provider unless it comes from the config. We have a way to do that, just check out our provisioning tests. I think this whole function plus another poll/wait that checks the workloads will be better in v2/validation/registries/registries.go. This function is too limiting to be an extension for the whole framework. Maybe you don't even need the for loop at all.

Reply:
Agreed, I think the rest of this file is pretty reusable except for this helper